server/call/session.py

```diff
@@ -436,11 +432,28 @@ def cancel_shutdown_timer(self):
         self._shutdown_timer.cancel()
         self._shutdown_timer = None
 
-    def get_auth_headers(self) -> dict[str, str]:
-        """Gets the authorization headers for Daily's REST API"""
+    def set_daily_auth_headers(self, room_url: str):
+        """Sets the Daily auth headers for this session, using either the default
+        API key or a domain-specific one if provided."""
+
+        # Default to primary API key
         api_key = self._config.daily_api_key
+
+        # If a room URL is provided, try to parse the subdomain
+        # and use a domain-specific API key
+        if room_url:
+            try:
+                parsed_url = urlparse(room_url)
+            except Exception as e:
+                raise Exception(
+                    f"Failed to parse room URL {room_url}") from e
+            subdomain = parsed_url.hostname.split('.')[0]
```

Comment:
Does this mean it'll only work for production domains, not staging or release?

Reply:
Yes! It will only work for production for now; though I can see value in supporting staging and release as well. I'd suggest adding that as a followup PR if we want.